src/components/structures/RoomView.tsx

```diff
@@ -1208,12 +1211,30 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
             case Action.EditEvent: {
                 // Quit early if we're trying to edit events in wrong rendering context
                 if (payload.timelineRenderingType !== this.state.timelineRenderingType) return;
+                if (payload.event && payload.event.getRoomId() !== this.state.roomId) {
+                    // If the event is in a different room, we need to view that room first
+                    dis.dispatch<ViewRoomPayload>({
+                        action: Action.ViewRoom,
+                        room_id: payload.even?.getRoomId(),
+                        metricsTrigger: "MessageSearch",
```

Comment:
why does an "edit" action cause a "MessageSearch" metricsTrigger?

Reply:
It doesn't inherently, just in this case, will omit it for risk of overdoing it